test/rotation.jl

```diff
@@ -0,0 +1,97 @@
+
```

Comment:
As Brian said, it would be good to make the tests device-aware. I suggest the following:
- Wrap your testsets in a function rotation_testsuite, taking the device (CPU, GPU) as argument
- All your current forward-pass tests against other libraries can be kept.
- If not too much work, modify the gradient tests to use `gradtest`, which should make them nice and short.
- If the device is CPU, run the tests as they are
- If the device is GPU, check the GPU results + gradients against the CPU results, which can be done automatically via gputest.

So essentially: 1) Check CPU against other libs 2) check CPU grads against ForwardDiff 3) check GPU forward and backward against CPU, if GPU is present. You don't have to copy the upsample tests 1:1, but you can draw inspiration from there.

Having good tests is essential to avoid headache later on, so stay strong! :D